app/src/main/java/de/tum/in/tumcampusapp/component/ui/cafeteria/model/CafeteriaMenu.kt

```diff
@@ -25,4 +27,63 @@ data class CafeteriaMenu(@PrimaryKey(autoGenerate = true)
                          var typeShort: String = "",
                          var typeLong: String = "",
                          var typeNr: Int = -1,
-                         var name: String = "")
\ No newline at end of file
+                         var name: String = "") {
+
+    private val formattedName: String
+        get() = COMPILE.matcher(name).replaceAll("").trim()
+
+    val menuType: MenuType
+        get() {
+            return when (typeShort) {
+                "tg" -> MenuType.DAILY_SPECIAL
+                "ae" -> MenuType.DISCOUNTED_COURSE
+                "akt" -> MenuType.SPECIALS
+                "bio" -> MenuType.BIO
+                else -> MenuType.SIDE_DISH
+            }
+        }
+
+    val notificationTitle: String
+        get() {
+            return PATTERN
+                    .matcher(typeLong)
+                    .replaceAll("")
+                    .trim()
+        }
+
+    fun getNotificationText(context: Context): String {
+        val lines = getNotificationLines(context)
+        return if (menuType == MenuType.SPECIALS) {
+            lines.joinToString(", ")
+        } else {
+            lines.first()
+        }
+    }
+
+    fun getNotificationLines(context: Context): List<String> {
+        return if (menuType == MenuType.SPECIALS) {
+            // Returns a list of all specials
+            formattedName
+                    .split("\n")
+                    .map { it.trim() }
+        } else {
+            // Returns a list containing the dish name and the price
+            val priceText = getPriceText(context)
+            listOfNotNull(formattedName, priceText)
+        }
+    }
+
+    private fun getPriceText(context: Context): String? {
+        val rolePrices = CafeteriaPrices.getRolePrices(context)
+        val price = rolePrices[typeLong]
+        return if (price != null) "$price €" else null
```

Comment:
I think this is a `price?.run(...)`

Reply:
Could you elaborate?

Reply:
```kotlin
return price?.run{"$price €"}
```

IMHO if you're writing `!= null` in Kotlin, you're doing it wrong...
Could just be me :man_shrugging:

Reply:
I like it. Will change accordingly.